services/common/src/redux/slices/complianceReportsSlice.ts

```diff
@@ -3,129 +3,197 @@ import { ENVIRONMENT } from "@mds/common/constants/environment";
-                    const records: IMineReportDefinition[] = action.payload.records ?? [];
-                    const { current_page, items_per_page, total, total_pages } = action.payload;
-                    state.reportPageData = { records, current_page, items_per_page, total, total_pages };
-                    state.params = action.meta.arg;
-                },
-                rejected: (state: ComplianceReportState, action) => {
-                    rejectHandler(action);
-                    // don't show loading forever if there's an error
-                    state.params = action.meta.arg;
-                }
-            }
-        )
-    }),
-    selectors: {
-        getMineReportDefinitionHash: (state): ItemMap<IMineReportDefinition> => {
-            return createItemMap(state.reportPageData.records, "mine_report_definition_guid")
+  name: complianceReportReducerType,
+  initialState,
+  reducers: (create) => ({
+    fetchComplianceReports: create.asyncThunk(
+      async (searchParams: ComplianceReportParams, thunkApi) => {
+        const headers = createRequestHeader();
+        thunkApi.dispatch(showLoading());
+        const resp = await CustomAxios({
+          errorToastMessage: "Failed to load compliance reports",
+        }).get(`${ENVIRONMENT.apiUrl}${API.MINE_REPORT_DEFINITIONS(searchParams ?? {})}`, headers);
+        thunkApi.dispatch(hideLoading());
+        return resp.data;
+      },
+      {
+        fulfilled: (state: ComplianceReportState, action) => {
+          const records: IMineReportDefinition[] = action.payload.records ?? [];
+          const { current_page, items_per_page, total, total_pages } = action.payload;
+          state.reportPageData = { records, current_page, items_per_page, total, total_pages };
+          state.params = action.meta.arg;
         },
-        getMineReportDefinitionOptions: (state): IMineReportDefinition[] => {
-            return state.reportPageData.records;
+        rejected: (state: ComplianceReportState, action) => {
+          rejectHandler(action);
+          // don't show loading forever if there's an error
+          state.params = action.meta.arg;
         },
-        getComplianceReportPageData: (state): IPageData<IMineReportDefinition> => {
-            return state.reportPageData;
+      }
+    ),
+    createMineReportDefinition: create.asyncThunk(
+      async (reportDefinition, thunkApi) => {
+        const headers = createRequestHeader();
+        thunkApi.dispatch(showLoading());
+        try {
+          const resp = await CustomAxios().post(
+            `${ENVIRONMENT.apiUrl}${API.MINE_REPORT_DEFINITION}`,
+            reportDefinition,
+            headers
+          );
+          return resp.data;
+        } catch (error) {
+          throw error;
+        } finally {
+          thunkApi.dispatch(hideLoading());
+        }
+      },
+      {
+        fulfilled: (state: ComplianceReportState, action) => {
+          notification.success({
```

Comment:
is it just me that prefers to use the successToastMessage of CustomAxios? It might be just me...

Reply:
I guess it's just more of aesthetic thing for me.  Since we're using a `fulfilled` function here, I always like to see all of the side efffects like that in one place.  It helps to prevent the "Why is this happening?" kind of issue.

That said, it's certainly not a hill I care to die on.  Both work.